### PR TITLE
Add track-to-track validation tool plugin. Added instances for general HLT track monitoring and in BTV specific paths 

### PR DESCRIPTION
**Pull request by johnalison**
Add track-to-track validation tool plugin. Added instances for general HLT track monitoring and in BTV specific paths #31038

PR description:
Adding plugin to compare two different track collections.
Creates relative efficiencies, fake rates, and resolutions

Configured track-to-track validator in the DQM monitor to monitor HLT tracks relative to offline tracks.
Two sets of monitoring plots configured, one with high-purity tracks and another with high-purity tracks from the PV.

Configured track-to-track validator in the BTVOffline

PR validation:
Standard runTheMaxtrix and look at the DQM plots produced

### CODE REVIEW

**Comment by johnalison** [comment]
Not sure what to do about the "files with invalid states" 
I removed these from the branch that is being merged.

**Comment by mtosi** [comment]
@johnalison thanks !
do you mind to update the module name and avoid the "validator" ?
the matching is based on dR between the 2 reco::Track, are you planning to enforce some reuqirements based on the hits as well ?

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by johnalison** [comment]
@mtosi  

"do you mind to update the module name and avoid the "validator" ?"
Sure,  any suggestions ?  TrackToTrackComparisonHists ?

For the hit-level matching, I still need to understand how to do it properly.  Once I have it I want to add histograms of the number of overlapping hits , relative hit-efficiency by layer, and an flag to do the track matching based on hit-level or dR. 

I figured b/c this was a bigger change I should start with what I have now and then do a PR later to add the hit-level stuff.

**Comment by mtosi** [comment]
@johnalison , thanks !

File: DQM/TrackingMonitorSource/interface/TrackToTrackComparisonHists.h [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  //std::vector<int> totSIM_vertz_fwdneg,totASS_vertz_fwdneg;
+  //std::vector<int> totREC_algo;
+  //std::vector<int> totREC_ootpu_entire, totASS2_ootpu_entire;
+  //std::vector<int> totREC_ootpu_barrel, totASS2_ootpu_barrel;
+  //std::vector<int> totREC_ootpu_fwdpos, totASS2_ootpu_fwdpos;
+  //std::vector<int> totREC_ootpu_fwdneg, totASS2_ootpu_fwdneg;
+  //std::vector<int> totREC_ootpu_eta_entire, totASS2_ootpu_eta_entire;
+  //std::vector<int> totASS2_itpu_eta_entire, totASS2_itpu_eta_entire_signal, totASS2_itpu_vertcount_entire, totASS2_itpu_vertcount_entire_signal;
+  //std::vector<int> totFOMT_eta, totFOMT_vertcount;
+  //std::vector<int> totCONeta, totCONvertcount, totCONzpos;
```
Comment (jfernan2) [inline DQM/TrackingMonitorSource/interface/TrackToTrackComparisonHists.h:165]: Could you please remove all this commented code? You can always access it by the commit history, Thanks

Comment (johnalison) [reply DQM/TrackingMonitorSource/interface/TrackToTrackComparisonHists.h]: Done

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@johnalison  added plots are empty, do you need by chance any specific HLT menu as in https://github.com/cms-sw/cmssw/pull/30366   ?
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_2_X_2020-08-07-1100+8e6f2f/38253/dqm-histo-comparison-summary.html

**Comment by johnalison** [comment]
The new tracking histograms have no explicit trigger requirement. 
11634.0 and 12423 look as expected

**Comment by jfernan2** [comment]
Right, I was referring to HLT/BTV/* all plots are empty

**Comment by johnalison** [comment]
The BTV plots are expected to be empty given the stringent trigger requirements. 
Ive tested locally with looser requirements to have test event passing and I see the selected histograms filled.

**Comment by mtosi** [comment]
which is the min stat needed for reasonable plots ?

**Comment by johnalison** [comment]
The paths are:
  HLT_Mu8_TrkIsoVVL_Ele23_CaloIdL_TrackIdL_IsoVL_DZ_PFDiJet30_PFBtagDeepCSV_1p5. and
  HLT_PFHT400_SixPFJet32_DoublePFBTagDeepCSV

so a few hundred ttbar should be enough.

**Comment by jfernan2** [comment]
@johnalison but if I get it right, in this other PR https://github.com/cms-sw/cmssw/pull/31104  you explicitely request the paths to be fired and indeed you reduce the stats, so this PR still should have some entires? I am confused:
https://tinyurl.com/y2rfqnor

**Comment by johnalison** [comment]
Yes good point. 

Let me get back to you.

**Comment by johnalison** [comment]
HI @jfernan2 

You are right that this has the same trigger requirement as PR #31104  however looking into the validation plots there more closely I indeed only see entries in the baseline histograms and not those from the PR.  The overlays you sent also only show entries in the blue reference with no trigger requirement. Im am running the the runTheMatrix locally  with 100 events to check if I see any passing triggers.   I also checked with a looser trigger to make sure all was OK before making the PR.

cheers (sorry for the late reply)

**Pull request merged**